vm/src/builtins/str.rs

```diff
@@ -826,10 +826,12 @@ impl PyStr {
         let s = self.as_str();
         match count {
             OptionalArg::Present(max_count) if max_count >= 0 => {
-                if max_count == 0 || s.is_empty() {
+                if max_count == 0 || (s.is_empty() && !old.is_empty()) {
                     // nothing to do; return the original bytes
-                    s.into()
-                } else {
+                    s.to_owned()
+                }else if s.is_empty() && old.is_empty(){
```

Comment:
running `cargo fmt --all` will fix this format warning

Reply:
Thanks for your advice! I modify that part by running `cargo fmt --all`